### PR TITLE
replace generic ID fields to more specific names

### LINKED ISSUE #1648: replace generic ID fields to more specific names
- [ ] UserOrderableServiceVo - `id` -> `serviceTemplateId` - https://github.com/eclipse-xpanse/xpanse/blob/f9754bf2013dfacdc773a60b417d7fdd6bd8a697/modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/view/UserOrderableServiceVo.java#L38
- [ ] UserPolicy and ServicePolicy- `userPolicyId` and `servicePolicyId` respectively.  https://github.com/eclipse-xpanse/xpanse/blob/82097e293b2f9c80146c31294619f1fd9a0b5798/modules/models/src/main/java/org/eclipse/xpanse/modules/models/policy/servicepolicy/ServicePolicy.java#L31
- [ ] ServiceTemplateDetailVo - serviceTemplateId - https://github.com/eclipse-xpanse/xpanse/blob/f9754bf2013dfacdc773a60b417d7fdd6bd8a697/modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/view/ServiceTemplateDetailVo.java#L43
- [ ] ServiceModificationAuditDetails - serviceModificationRequestId https://github.com/eclipse-xpanse/xpanse/blob/8fb4e801b728ebb81afacb7af17683e0d0fda7a3/modules/models/src/main/java/org/eclipse/xpanse/modules/models/service/modify/ServiceModificationAuditDetails.java#L31
- [ ] MigrateRequest - originalServiceId https://github.com/eclipse-xpanse/xpanse/blob/8475effe6e79b1ac95b4984e58dc38986fc0f547/modules/models/src/main/java/org/eclipse/xpanse/modules/models/workflow/migrate/MigrateRequest.java#L33
- [ ] DeployedService - serviceId https://github.com/eclipse-xpanse/xpanse/blob/8fb4e801b728ebb81afacb7af17683e0d0fda7a3/modules/models/src/main/java/org/eclipse/xpanse/modules/models/service/view/DeployedService.java#L34

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Fixed #1648
related to eclipse-xpanse/xpanse-ui/pull/858

### CODE REVIEW

**Comment by baixinsui** [comment]
LGTM

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/migration/steps/StartDeploy.java [hunk 1]
```diff
@@ -84,6 +84,7 @@ private void startDeploy(String processInstanceId, UUID newServiceId,
                 (MigrateRequest) variables.get(MigrateConstants.MIGRATE_REQUEST);
         DeployRequest deployRequest = new DeployRequest();
         BeanUtils.copyProperties(migrateRequest, deployRequest);
+        deployRequest.setId(migrateRequest.getOriginalServiceId());
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/migration/steps/StartDeploy.java:87]: we can update the Id filed in DeployRequest also as 'ServiceId'

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/ServiceTemplateEntityConverter.java [hunk 2]
```diff
@@ -34,6 +34,7 @@ public static ServiceTemplateDetailVo convertToServiceTemplateDetailVo(
                 && Objects.nonNull(serviceTemplateEntity.getOcl())) {
             ServiceTemplateDetailVo serviceTemplateDetailVo = new ServiceTemplateDetailVo();
             BeanUtils.copyProperties(serviceTemplateEntity, serviceTemplateDetailVo);
+            serviceTemplateDetailVo.setServiceTemplateId(serviceTemplateEntity.getId());
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/config/ServiceTemplateEntityConverter.java:37]: Does it make sense to also to rename the ID field in the entities?

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Comment by Alice1319** [comment]
LGTM

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**